app/Roster/Roster/RosterQueryTrait.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+
+namespace App\Roster;
+
+
+use App\Roster;
+use App\TableProperties;
+use Illuminate\Database\Eloquent\Relations\BelongsToMany;
+
+trait RosterQueryTrait
+{
+
+    public function rosters()
+    {
+        return $this->hasMany(Roster::class, $this->foriegnPivotKey());
+    }
+
+    /**
+     * @return BelongsToMany
+     */
+    public function players()
+    {
+        return $this->rosters()->with('player')->ofPlayerType();
+    }
+
+    /**
+     * @return BelongsToMany
+     */
+    public function proficiencies()
+    {
+        return  $this->rosters()->with('proficiency');
```

Comment:
proficiency_id is nullable, fix it with 
`return  $this->rosters()->whereNotNull('proficiency_id')`